### PR TITLE
DBName (--db-name) cli flag updates

### PR DESCRIPTION
**Pull request by manthanhd**

### CODE REVIEW

**Comment by manthanhd** [comment]
For some reason on my local it has stopped picking up --build-dir flag. Not sure if this is an issue specific to me or something else.

File: cmd/startnode.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func init() {
 	StartnodeCmd.Flags().StringVar(&flags.BootstrapIDs, "bootstrap-ids", flags.BootstrapIDs, "Comma separated list of bootstrap peer ids to connect to. Example: NodeID-JR4dVmy6ffUGAKCBDkyCbeZbyHQBeDsET,NodeID-8CrVPQZ4VSqgL8zTdvL14G8HqAfrBr4z")
 	StartnodeCmd.Flags().StringVar(&flags.BootstrapBeaconConnectionTimeout, "bootstrap-beacon-connection-timeout", flags.BootstrapBeaconConnectionTimeout, "Timeout when attempting to connect to bootstrapping beacons.")
 
-	StartnodeCmd.Flags().BoolVar(&flags.DBEnabled, "db-enabled", flags.DBEnabled, "Turn on persistent storage.")
+	StartnodeCmd.Flags().StringVar(&flags.DBType, "db-type", flags.DBType, "Name of the DB to use.")
```
Comment (aaronbuchwald) [inline cmd/startnode.go:136]: Can you update the comment to `Database type to use.`

File: node/cli_tools.go [hunk 2]
```diff
@@ -67,7 +68,7 @@ func FlagsToArgs(flags Flags, basedir string, sepBase bool) ([]string, Metadata)
 		"--bootstrap-ips=" + flags.BootstrapIPs,
 		"--bootstrap-ids=" + flags.BootstrapIDs,
 		"--bootstrap-beacon-connection-timeout=" + flags.BootstrapBeaconConnectionTimeout,
-		"--db-enabled=" + strconv.FormatBool(flags.DBEnabled),
+		"--db-type=" + dbType,
```
Comment (aaronbuchwald) [inline node/cli_tools.go:71]: This needs to be `flags.DBType`, so that if a user specifies the database type, it will be passed in correctly.

**Comment by manthanhd** [comment]
Needs to be merged with avalanche-internal changes for DBType

File: node/config.go [hunk 3]
```diff
@@ -190,7 +187,7 @@ type FlagsYAML struct {
 	BootstrapIPs                            *string  `yaml:"bootstrap-ips,omitempty"`
 	BootstrapIDs                            *string  `yaml:"bootstrap-ids,omitempty"`
 	BootstrapBeaconConnectionTimeout        *string  `yaml:"bootstrap-beacon-connection-timeout,omitempty"`
-	DBEnabled                               *bool    `yaml:"db-enabled,omitempty"`
+	DBType                                  *bool    `yaml:"db-type,omitempty"`
```
Comment (aaronbuchwald) [inline node/config.go:190]: Shouldn't this be a string instead of a boolean?

Comment (manthanhd) [reply node/config.go]: wow can't believe I missed that. I guess it was working when I was running it locally because I don't use the yaml based config.

File: cmd/startnode.go [hunk 4]
```diff
@@ -53,8 +53,10 @@ var StartnodeCmd = &cobra.Command{
 		}
 
 		args, md := node.FlagsToArgs(flags, sanitize.Path(datapath), false)
-		// Set flags to default for next `startnode` call
-		flags = node.DefaultFlags()
+		defer func() {
+			// Set flags to default for next `startnode` call
+			flags = node.DefaultFlags()
+		}()
```
Comment (aaronbuchwald) [inline cmd/startnode.go:59]: What's the purpose of this change?

**Comment by aaronbuchwald** [comment]
Changing the base to dev, since we do not want to merge this to master until the corresponding flag change is released on avalanchego.

File: cmd/startnode.go (hunk 4 above)
Comment (manthanhd) [reply cmd/startnode.go:59]: without it the node.DefaultFlags was overwriting the data in flags before using it.

**Review by aaronbuchwald — APPROVED** [review]
LGTM
